Features/music selector

mercenaire opened the pull request:
The music selection feature is ready.
N = play next track
S = shuffle mode on/off
L = loop single track on/off
Files should be dumped into data/music (as configured in Config.pde)
Acceptable filetypes: only *.mp3 for now (as configured in MusicLibrary.pde)

Commit:
Initial implementation of music selector. N=next track, S=toggle shuffle, L=toggle loop single track.

Commit:
Documented MusicLibrary; fixed crash with extensionless files/directories in the music directory.

Commit:
Fix bug when there is only one music file.

scjody commented:
whoo, :musical_note:!  I'll take a look at this PR tomorrow unless @mgoimam beats me to it!

mgoimam committed:
Fix audio issues

- Don't arbitrarily reduce the gain of the music. Instead,
  we expect audio to be balanced manually.
- Don't stop the whole minim library when stopping the music.

scjody commented:
I only took a quick look but this looks great, thanks!  Feel free to merge once @mgoimam is happy.

Commit:
Further audio issue fixes as per mgoimam code review:
- Close the currently active audio player instead of the whole library
- Fix the repeat-avoidance part of the shuffle algorithm
